### DISCUSSION TITLE
Input (Switch,Button) state on website

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#16995] [Q&A]
Hi, why there is no nativ way to show state of inputs on WebUI?
There should be a reason that i don't see.

I want to avoid binding a gpio to a (virtual) relay, to get the state shown on webui.
What is the best workaround? 

I have set two Relays(#1&2) as digital outputs (open garage & close garage) and
and one switch (#3) as input (garage is closed).
How to show the state of the switch on webui?

thx!

**Comment by sfromis** [comment]
The general idea with the very rudimentary Tasmota web UI is that it is not meant as a "device dashboard", more for admin purposes, and test of a few of the features. Hence infrastructure for being a dashboard was not included (very memory-constrained chip), and the slow update rate would anyway not work well for constantly updated state of pins. The structure is more, as the "M" in the Tasmota name suggests, to connect to a backend dashboard via MQTT, where you can manage state and actions for multiple devices. For testing purposes, you can also have state changes appear in the console using rules, of course this was not what you asked for, but again, the web UI front page is just not meant for what you'd like.

On ESP32 you have Berry scripting with quite a few features where you can add things to the web UI, and the very different scripting language available with ESP8266 also can do something, with some learning curve.

**Comment by ChrisVonFrosta** [comment]
i know what u mean.
But the Tamsota web UI shows nearly everything else. 
Buttons for relays and sensor values.
So i think the option to show inputs states would not brake the paradigm,  even just for test purposes .

anyway, thx for answer!
